src/sagemaker/jumpstart/cache.py

```diff
@@ -129,15 +149,44 @@ def get_region(self) -> str:
         """Return region for cache."""
         return self._region
 
-    def set_manifest_file_s3_key(self, key: str) -> None:
-        """Set manifest file s3 key. Clears cache after new key is set."""
-        if key != self._manifest_file_s3_key:
-            self._manifest_file_s3_key = key
+    def set_manifest_file_s3_key(
+        self,
+        key: str,
+        file_type: JumpStartS3FileType = JumpStartS3FileType.OPEN_WEIGHT_MANIFEST,
+    ) -> None:
+        """Set manifest file s3 key, clear cache after new key is set.
+
+        Raises:
+            ValueError: if the file type is not recognized
+        """
+        file_mapping = {
+            JumpStartS3FileType.OPEN_WEIGHT_MANIFEST: self._manifest_file_s3_key,
+            JumpStartS3FileType.PROPRIETARY_MANIFEST: self._proprietary_manifest_s3_key,
+        }
+        property_name = file_mapping.get(file_type)
+        if not property_name:
+            raise ValueError(
+                f"Bad value when setting manifest '{file_type}': must be in"
+                f" {JumpStartS3FileType.OPEN_WEIGHT_MANIFEST}"
+                f" {JumpStartS3FileType.PROPRIETARY_MANIFEST}"
+            )
+        if key != property_name:
+            setattr(self, property_name, key)
             self.clear()
 
-    def get_manifest_file_s3_key(self) -> str:
+    def get_manifest_file_s3_key(
+        self, file_type: JumpStartS3FileType = JumpStartS3FileType.OPEN_WEIGHT_MANIFEST
+    ) -> str:
         """Return manifest file s3 key for cache."""
-        return self._manifest_file_s3_key
+        if file_type == JumpStartS3FileType.OPEN_WEIGHT_MANIFEST:
+            return self._manifest_file_s3_key
+        if file_type == JumpStartS3FileType.PROPRIETARY_MANIFEST:
+            return self._proprietary_manifest_s3_key
+        raise ValueError(
+            f"Bad value when getting manifest '{file_type}':"
+            f"must be in {JumpStartS3FileType.OPEN_WEIGHT_MANIFEST}"
+            f"{JumpStartS3FileType.PROPRIETARY_MANIFEST}"
```

Comment:
careful, I think you are missing spaces, it would print `:must be in <val1><val2>`

Verify in unit test please

Reply:
Thanks for catching! yes fixed